girder/api/v1/item.py

```diff
@@ -172,7 +175,32 @@ def setMetadata(self, item, metadata, params):
                 raise RestException(
                     'Invalid key %s: keys must not start with the "$" character.' % k)
 
-        return self.model('item').setMetadata(item, metadata)
+        return self.model('item').setMetadata(item, metadata, params['allowNull'])
+
+    @autoDescribeRoute(
+        Description('Delete metadata fields on an item.')
+        .responseClass('Item')
+        .modelParam('id', model='item', level=AccessType.WRITE)
+        .jsonParam('fields', 'A JSON list containing the metadata fields to delete',
+                   paramType='body')
```

Comment:
If this `fields` param is expected to be a JSON array, let's add `requireArray=True`. We could validate even further by passing a `schema` dict to indicate it's an Array of Strings and may not be empty.